crates/tabby/src/serve/mod.rs

```diff
@@ -201,44 +201,55 @@ fn api_router(args: &ServeArgs) -> Router {
         None
     };
 
+    let mut routers = vec![];
+
     let health_state = Arc::new(health::HealthState::new(args));
-    let router = Router::new()
-        .route("/v1/events", routing::post(events::log_event))
-        /* Remove POST /v1/health route in next major version release. */
-        .route(
-            "/v1/health",
-            routing::post(health::health).with_state(health_state.clone()),
-        )
-        .route(
-            "/v1/health",
-            routing::get(health::health).with_state(health_state),
-        )
-        .route(
-            "/v1/completions",
-            routing::post(completions::completions).with_state(completion_state),
-        );
+    routers.push({
+        Router::new()
+            .route("/v1/events", routing::post(events::log_event))
+            .route(
+                "/v1/health",
+                routing::post(health::health).with_state(health_state.clone()),
+            )
+            .route(
+                "/v1/health",
+                routing::get(health::health).with_state(health_state),
+            )
+    });
 
-    let router = if let Some(chat_state) = chat_state {
-        router.route(
-            "/v1beta/chat/completions",
-            routing::post(chat::completions).with_state(chat_state),
-        )
-    } else {
-        router
-    };
+    routers.push({
+        Router::new()
+            .route(
+                "/v1/completions",
+                routing::post(completions::completions).with_state(completion_state),
+            )
+            .layer(TimeoutLayer::new(Duration::from_secs(3)))
```

Comment:
All completion request have a timeout limit of 3s, so this means there is no option to manually trigger and wait for a completion request with a longer response time?

Reply:
In another word, even it's a manual completion, we should still provide SLA <= 3s